landing-page/content/common/index-and-statistics-format.md

```diff
@@ -0,0 +1,126 @@
+
+A file conforming to the format specification should have the structure as
+described below.
+
+### File structure
+
+The file has the following structure
+
+```
+Magic Blob₁ Blob₂ ... Blobₙ Footer
+```
+
+where
+
+- `Magic` is four bytes 0x50, 0x46, 0x49, 0x53 (short for: Plain Format for
+  Indices and Statistics),
+- `Blobᵢ` is i-th blob contained in the file, to be interpreted by application
+  according to the footer,
+- `Footer` is defined below.
+
+### Footer structure
+
+Footer has the following structure
+
+```
+Magic FooterPayload FooterPayloadSize Reserved Flags FileFormatVersion Magic
+```
+
+where
+
+- `Magic` four bytes, same as at the beginning of the file.
+- `FooterPayload` optionally compressed, UTF-8 encoded JSON payload describing the
+  blobs in the file, with the structure described below,
+- `FooterPayloadSize` is a length in bytes of the `FooterPayload` (compressed),
+  stored as 4 byte integer, little-endian,
+- `Reserved` is 4 bytes reserved for future use, currently should be written as
+  0x00, 0x00, 0x00, 0x00,
+- `Flags` a 4 byte integer, stored little-endian, for boolean flags
+  - 0 (lowest bit): whether `FooterPayload` is compressed
+  - all other bits are reserved for future use and should be set to 0 on write
+- `FileFormatVersion` is a number, stored as 4 byte integer, little-endian,
+
+### Footer Payload
+
+Footer payload bytes is either uncompressed or LZ4-compressed (as a single LZ4
+compression frame with content size present), UTF-8 encoded JSON payload
+representing a single `FileMetadata` object.
+
+#### FileMetadata
+
+`FileMetadata` has the following fields
+
+
+| Field Name | Field Type                              | Required | Description |
+| ---------- | --------------------------------------- | -------- | ----------- |
+| blobs      | list of BlobMetadata objects            | yes      |
+| properties | JSON object with string property values | no       | storage for arbitrary meta-information, like writer identification/version
+
+#### BlobMetadata
+
+`BlobMetadata` has the following fields
+
+| Field Name | Field Type             | Required | Description |
+| ---------- | ---------------------- | -------- | ----------- |
+| type       | JSON string            | yes      | See [Blob types](#blob-types)
+| columns    | list of JSON long      | yes      | list of column IDs the blob was computed for
+| offset     | JSON long              | yes      | The offset in the file where the blob contents start. Reader should assume the value can be more than 2^32.
+| length     | JSON long              | yes      | The length of the blob stored in the file
+| compression_codec | JSON string     | no       | See [Compression codecs](#compression-codecs). If omitted, the data is assumed to be uncompressed.
+
+### Blob types
+
+The blobs can be of a type listed below
+
+#### `ndv-long-little-endian` blob type
+
+8-bytes integer stored little-endian and representing number of distinct values.
```

Comment:
signed or unsigned?

Reply:
unsigned. will add here. Although i expect applications not to support values beyond 2^63-1

Reply:
Making it signed requires me to say "two's complement", which is unnecessary complexity for what is a non-negative number. 

Note that I took a different approach for integer numbers in the footer, because footer _may_ gain some integer fields that can be negative, and having to deal with unsigned and signed numbers there would be impractical.

Here, the definition is well-isolated, independent, so I it's actually simpler to define it as unsigned.

Reply:
SGTM, we should probably state the constraint in the specification even if it is unlikely to be hit in practice (i.e. it sounds like any value `>  2^63-1` should be capped at that value).

Reply:
The capping -- if any is needed -- should be applied on the reader side, it doesn't belong to the specification.
(Unless we really want to say that values bigger than 2^63-1 are disallowed. I didn't find it the right thing to do)

Reply:
Ah, sorry I misunderstood.  I misunderstood your original reply.  So it is expected to be the full range of values.